rustler_codegen/src/tagged_enum.rs

```diff
@@ -0,0 +1,256 @@
+use std::collections::HashMap;
+
+use proc_macro2::{Span, TokenStream};
+use quote::{quote, quote_spanned};
+
+use heck::ToSnakeCase;
+use syn::{self, spanned::Spanned, Fields, Ident, Variant};
+
+use super::context::Context;
+
+pub fn transcoder_decorator(ast: &syn::DeriveInput) -> TokenStream {
+    let ctx = Context::from_ast(ast);
+
+    let variants = ctx
+        .variants
+        .as_ref()
+        .expect("NifEnum can only be used with enums");
+
+    // Remove duplicated atoms.
+    let atom_set = variants
+        .iter()
+        .flat_map(|variant| {
+            let mut ret: Vec<(String, Ident)> = if let Fields::Named(fields) = &variant.fields {
+                fields
+                    .named
+                    .iter()
+                    .map(|field| {
+                        let atom_str = field.ident.as_ref().expect("").to_string().to_snake_case();
```

Comment:
Shouldn't we have an error message for the `expect("")` here?

Maybe:
```suggestion
                        let atom_str = field.ident.as_ref().expect("Atom is expected").to_string().to_snake_case();
```